src/librustc/middle/stability.rs

```diff
@@ -320,6 +415,26 @@ impl<'a, 'tcx> Visitor<'tcx> for Annotator<'a, 'tcx> {
     fn visit_macro_def(&mut self, md: &'tcx hir::MacroDef) {
         self.annotate(md.hir_id, &md.attrs, md.span, AnnotationKind::Required, |_| {});
     }
+
+    fn visit_generic_param(&mut self, gp: &'tcx hir::GenericParam) {
+        let kind = match &gp.kind {
+            hir::GenericParamKind::Lifetime {..} => AnnotationKind::Prohibited,
+            hir::GenericParamKind::Type {
+                default,
+                ..
+            } => if default.is_some() {
+                AnnotationKind::Required
```

Comment:
I'm actually surprised you don't get a load of errors on generic parameters inside the standard library that are not marked as `#[stable]` or `#[unstable]`.

Reply:
I believe the reason is that I'm not inheriting stability and deprecations, prior to that there were a lot of those types of errors.